<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SNPassoc">
<title>SNPassoc: an R package to perform whole genome association studies • SNPassoc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SNPassoc: an R package to perform whole genome association studies">
<meta property="og:description" content="SNPassoc">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SNPassoc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0-17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/SNPassoc.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/brgelab" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/isglobal-brge/SNPassoc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>SNPassoc: an R package to perform whole genome association studies</h1>
                        <h4 data-toc-skip class="author">Juan R.
Gonzalez</h4>
            <address class="author_afil">
      Bioinformatics Research Group in Epidemiolgy (BRGE), Barcelona
Institute for Global Health (ISGlobal)Department of Mathematics,
Autonomous University of Barcelona
(UAB)<br><a class="author_email" href="mailto:#"></a><a href="mailto:juanr.gonzalez@isglobal.org" class="email">juanr.gonzalez@isglobal.org</a>
      </address>
                              <h4 data-toc-skip class="author">Victor
Moreno</h4>
            <address class="author_afil">
      Catalan Institute of Oncology<br><h4 data-toc-skip class="date">2022-10-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/isglobal-brge/SNPassoc/blob/HEAD/vignettes/SNPassoc.Rmd" class="external-link"><code>vignettes/SNPassoc.Rmd</code></a></small>
      <div class="d-none name"><code>SNPassoc.Rmd</code></div>
    </address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>SNPassoc</code> package contains facilities for data
manipulation, tools for exploratory data analysis, convenient graphical
facilities, and tools for assessing genetic association for both
quantitative and categorial (case-control) traits in whole genome
approaches. Genome-based studies are normally analyzed using a
multistage approach. In the first step researchers are interested in
assessing association between the outcome and thousands of SNPs. In a
second and possibly third step, medium/large scale studies are performed
in which only a few hundred of SNPs, those with a putative association
found in the first step, are genotyped. <code>SNPassoc</code> is
specially designed for analyzing this kind of designs. In addition, a
convenience-based approach (select variants on the basis of logistical
considerations such as the ease and cost of genotyping) can also be
analyzed using <code>SNPassoc</code>. Different genetic models are also
implemented in the package. Analysis of multiple SNPs can be analyzed
using either haplotype or gene-gene interaction approaches.</p>
<p>This document is an updated version of the initial vignette that was
published with the SNPassoc paper <span class="citation">González et al.
(2007)</span>. It contains a more realistic example belonging to a real
dataset. The original vignette is still available <a href="https://github.com/isglobal-brge/TeachingMaterials/blob/master/Master_Bioinformatics/Recommended_lectures/SupplementaryMaterial-BIOINF-2006-1602.R1.pdf" class="external-link">here</a>.</p>
</div>
<div class="section level2">
<h2 id="data-loading">Data loading<a class="anchor" aria-label="anchor" href="#data-loading"></a>
</h2>
<p>SNP data are typically available in text format or Excel spreadsheets
which are easily uploaded in <code>R</code> as a data frame. Here, as an
illustrative example, we are analyzing a dataset containing
epidemiological information and 51 SNPs from a case-control study on
asthma. The data is available within <code>SNPassoc</code> and can be
loaded by</p>
<p>Then, the data is loaded into the R session by</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">asthma</span>, package <span class="op">=</span> <span class="st">"SNPassoc"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">asthma</span>, list.len<span class="op">=</span><span class="fl">9</span><span class="op">)</span></span></code></pre></div>
<pre><code>'data.frame':   1578 obs. of  57 variables:
 $ country    : Factor w/ 10 levels "Australia","Belgium",..: 5 5 5 5 5 5 5 5 5 5 ...
 $ gender     : Factor w/ 2 levels "Females","Males": 2 2 2 1 1 1 1 2 1 1 ...
 $ age        : num  42.8 50.2 46.7 47.9 48.4 ...
 $ bmi        : num  20.1 24.7 27.7 33.3 25.2 ...
 $ smoke      : int  1 0 0 0 0 1 0 0 0 0 ...
 $ casecontrol: int  0 0 0 0 1 0 0 0 0 0 ...
 $ rs4490198  : Factor w/ 3 levels "AA","AG","GG": 3 3 3 2 2 2 3 2 2 2 ...
 $ rs4849332  : Factor w/ 3 levels "GG","GT","TT": 3 2 3 2 1 2 3 3 2 1 ...
 $ rs1367179  : Factor w/ 3 levels "CC","GC","GG": 2 2 2 3 3 3 2 3 3 3 ...
  [list output truncated]</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">asthma</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span></span></code></pre></div>
<pre><code>  country  gender      age      bmi smoke casecontrol rs4490198 rs4849332
1 Germany   Males 42.80630 20.14797     1           0        GG        TT
2 Germany   Males 50.22861 24.69136     0           0        GG        GT
3 Germany   Males 46.68857 27.73230     0           0        GG        TT
4 Germany Females 47.86311 33.33187     0           0        AG        GT
5 Germany Females 48.44079 25.23634     0           1        AG        GG</code></pre>
<p>We observe that we have case-control status (0: control, 1: asthma)
and another 4 variables encoding the country of origin, gender, age,
body mass index (bmi) and smoking status (0: no smoker, 1: ex-smoker, 2:
current smoker). There are 51 SNPs whose genotypes are given by the
alleles names.</p>
</div>
<div class="section level2">
<h2 id="descriptive-analysis">Descriptive analysis<a class="anchor" aria-label="anchor" href="#descriptive-analysis"></a>
</h2>
<p>To start the analysis, we must indicate which columns of the dataset
<code>asthma</code> contain the SNP data, using the
<code>setupSNP</code> function. In our example, SNPs start from column 7
onwards, which we specify in argument <em>colSNPs</em></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/isglobal-brge/SNPassoc" class="external-link">SNPassoc</a></span><span class="op">)</span></span>
<span><span class="va">asthma.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setupSNP.html">setupSNP</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">asthma</span>, colSNPs<span class="op">=</span><span class="fl">7</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p>This is an alternative way of determining the columns containing the
SNPs</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^rs"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">asthma.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setupSNP.html">setupSNP</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">asthma</span>, colSNPs<span class="op">=</span><span class="va">idx</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p>The argument <em>sep</em> indicates the character separating the
alleles. The default value is ’‘/´´. In our case, there is no separating
character, so that, we set <em>sep=““</em>. The argument
<em>name.genotypes</em> can be used when genotypes are available in
other formats, such as 0, 1, 2 or’‘norm´´,’‘het´´,’’mut´´. The purpose
of the <code>setupSNP</code> function is to assign the class
<em>snp</em> to the SNPs variables, to which <code>SNPassoc</code>
methods will be applied. The function labels the most common genotype
across subjects as the reference one. When numerous SNPs are available,
the function can be parallelized through the argument <em>mc.cores</em>
that indicates the number of processors to be used. We can verify that
the SNP variables are given the new class <em>snp</em></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">$</span><span class="va">rs1422993</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] G/G G/T G/G G/T G/T G/G
Genotypes: G/G G/T T/T
Alleles:  G T </code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">$</span><span class="va">rs1422993</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "snp"    "factor"</code></pre>
<p>and summarize their content with <code>summary</code></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">$</span><span class="va">rs1422993</span><span class="op">)</span></span></code></pre></div>
<pre><code>Genotypes: 
    frequency percentage
G/G       903  57.224335
G/T       570  36.121673
T/T       105   6.653992

Alleles: 
  frequency percentage
G      2376   75.28517
T       780   24.71483

HWE (p value): 0.250093 </code></pre>
<p>which shows the genotype and allele frequencies for a given SNP,
testing for Hardy-Weinberg equilibrium (HWE). We can also visualize the
results in a plot by</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">$</span><span class="va">rs1422993</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="SNPassoc_files/figure-html/plotsummarySNP-1.png" alt="SNP summary. Bar chart showing the basic information of a given SNP" width="700"><p class="caption">
SNP summary. Bar chart showing the basic information of a given SNP
</p>
</div>
<p>The argument <em>type</em> helps to get a pie chart</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">$</span><span class="va">rs1422993</span>, type<span class="op">=</span><span class="va">pie</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="SNPassoc_files/figure-html/plotsummarySNP2-1.png" alt="SNP summary. Pie chart showing the basic information of a given SNP" width="700"><p class="caption">
SNP summary. Pie chart showing the basic information of a given SNP
</p>
</div>
<p>The <em>summary</em> function can also be applied to the whole
dataset</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">asthma.s</span>, print<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>           alleles major.allele.freq HWE      missing (%)
rs4490198  A/G     59.2              0.174133  0.6       
rs4849332  G/T     61.8              0.522060  0.1       
rs1367179  G/C     81.4              0.738153  1.0       
rs11123242 C/T     81.7              0.932898  0.6       
rs13014858 G/A     58.3              0.351116  0.1       
rs1430094  G/A     66.9              0.305509  0.4       
rs1430093  C/A     66.6              0.817701  3.5       
rs746710   G/C     51.5              0.614368  0.0       
rs1430090  T/G     70.0              0.025180  1.6       
rs6737251  C/T     69.3              0.235996  0.3       
rs11685217 C/T     80.1              0.009462  4.5       
rs1430097  C/A     65.1              0.738166  1.0       
rs10496465 A/G     85.8              0.917997  0.6       
rs3756688  T/C     63.9              0.154632  0.6       
rs2303063  A/G     53.0              0.722069  1.1       
rs1422993  G/T     75.3              0.250093  0.0       
rs2400478  G/A     62.6              0.256786  0.9       
rs714588   A/G     54.9              0.838329  0.8       
rs1023555  T/A     76.8              0.943443  0.5       
rs898070   G/A     62.6              1.000000  0.6       
rs963218   C/T     53.2              0.389387  0.3       
rs1419835  C/T     78.2              0.505391  0.6       
rs765023   T/C     64.5              0.030513  6.9       
rs1345267  A/G     61.0              0.112183  0.1       
rs324381   G/A     64.6              0.242223 11.6       
rs184448   T/G     55.9              0.008446  2.2       
rs324396   C/T     71.2              0.197291  0.3       
rs324957   G/A     57.0              0.007417  0.4       
rs324960   C/T     66.6              0.077777  1.1       
rs10486657 C/T     81.3              0.672703  4.3       
rs324981   A/T     53.2              0.048438  0.2       
rs1419780  C/G     80.8              0.569652  0.2       
rs325462   T/A     51.0              0.337862  0.3       
rs727162   G/C     78.5              0.765708  0.0       
rs10250709 G/A     65.4              0.266434  0.0       
rs6958905  T/C     64.7              0.377472  0.4       
rs10238983 T/C     75.7              0.216435  0.4       
rs4941643  A/G     54.1              0.635887  7.2       
rs3794381  C/G     71.7              0.652089  7.1       
rs2031532  G/A     65.0              0.911918  0.0       
rs2247119  T/C     71.5              0.457710  0.5       
rs8000149  T/C     63.2              0.588077  0.4       
rs2274276  G/C     57.0              0.571386  0.6       
rs7332573  G/T     91.5              0.869947  1.5       
rs3829366  T/A     51.6              0.722626  1.3       
rs6084432  G/A     83.7              0.266716  0.6       
rs512625   G/A     69.5              0.905395  0.4       
rs3918395  G/T     86.8              0.508732  1.2       
rs2787095  G/C     60.2              0.102053  0.8       
rs2853215  G/A     73.0              0.249516  0.2       </code></pre>
<p>showing the SNP labels with minor/major allele format, the major
allele frequency the HWE test and the percentage of missing genotypes.
Missing values can be further explored plotting with</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotMissing.html">plotMissing</a></span><span class="op">(</span><span class="va">asthma.s</span>, print.labels.SNPs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="SNPassoc_files/figure-html/plotMissing-1.png" alt="Missing genotypes. Black squares shows missing genotuype information of asthma data example." width="700"><p class="caption">
Missing genotypes. Black squares shows missing genotuype information of
asthma data example.
</p>
</div>
<p>This plot can be used to inspect if missing values appear randomly
across individuals and SNPs. In our case, we can see that the missing
pattern may be considered random, except for three clusters in
consecutive SNPs (large black squares). These individuals should be
further checked for possible problems with genotyping.</p>
</div>
<div class="section level2">
<h2 id="hardy-weinberg-equilibrium">Hardy-Weinberg equilibrium<a class="anchor" aria-label="anchor" href="#hardy-weinberg-equilibrium"></a>
</h2>
<p>Genotyping of SNPs needs to pass quality control measures. Aside from
technical details that need to be considered for filtering SNPs with low
quality, genotype calling error can be detected by a HWE test. The test
compares the observed genotype frequencies with those expected under
random mating, which follows when the SNPs are in the absence of
selection, mutation, genetic drift, or other forces. Therefore, HWE must
be checked only in controls. There are several tests described in the
literature to verify HWE. In <code>SNPassoc</code> HWE is tested for all
the bi-allelic SNP markers using a fast exact test <span class="citation">Wigginton, Cutler, and Abecasis (2005)</span>
implemented in the <em>tableHWE</em> function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tableHWE.html">tableHWE</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hwe</span><span class="op">)</span></span></code></pre></div>
<pre><code>           HWE (p value)
rs4490198      0.1741325
rs4849332      0.5220596
rs1367179      0.7381531
rs11123242     0.9328981
rs13014858     0.3511162
rs1430094      0.3055089</code></pre>
<p>We observe that the first SNPs in the dataset are under HWE since
their P-values rejecting the HWE hypothesis (null hypothesis) are larger
than 0.05. However, when tested in control samples only, by stratifying
by cases and controls</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hwe2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tableHWE.html">tableHWE</a></span><span class="op">(</span><span class="va">asthma.s</span>, <span class="va">casecontrol</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#SNPs is HWE in the whole sample but not controls</span></span>
<span><span class="va">snpNHWE</span> <span class="op">&lt;-</span> <span class="va">hwe2</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">&gt;</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">hwe2</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">&lt;</span><span class="fl">0.05</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hwe2</span><span class="op">)</span><span class="op">[</span><span class="va">snpNHWE</span><span class="op">]</span></span></code></pre></div>
<pre><code>[1] "rs1345267"</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hwe2</span><span class="op">[</span><span class="va">snpNHWE</span>,<span class="op">]</span></span></code></pre></div>
<pre><code>all groups          0          1 
0.11218285 0.04956349 0.81604706 </code></pre>
<p>We see that rs1345267 is not in HWE within controls because its
P-value is &lt;0.05. Notice that one is interested in keeping those
SNPsthat do not reject the null hypothesis. As several SNPs are tested,
multiple comparisons must be considered. In this particular setting, a
threshold of 0.001 is normally considered. As a quality control measure,
it is not necessary to be as conservative as in those situations where
false discovery rates need to be controlled.</p>
<p>SNPs that do not pass the HWE test must be removed form further
analyses. We can recall <em>setupSNP</em> and indicate the columns of
the SNPs to be kept</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snps.ok</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">hwe2</span><span class="op">)</span><span class="op">[</span><span class="va">hwe2</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">&gt;=</span><span class="fl">0.001</span><span class="op">]</span></span>
<span><span class="va">pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">snps.ok</span>, useNames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">asthma.s</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setupSNP.html">setupSNP</a></span><span class="op">(</span><span class="va">asthma</span>, <span class="va">pos</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p>Note that in the variable <em>pos</em> we redefine the SNP variables
to be considered as class <em>snp</em>.</p>
</div>
<div class="section level2">
<h2 id="snp-association-analysis">SNP association analysis<a class="anchor" aria-label="anchor" href="#snp-association-analysis"></a>
</h2>
<p>We are interested in finding those SNPs associated with asthma status
that is encoded in the variable <em>casecontrol</em>. We first
illustrate the association between case-control status and the SNP
rs1422993. The association analysis for all genetic models is performed
by the function <em>association</em> that regresses <em>casecontrol</em>
on the variable rs1422993 from the dataset <em>asthma.s</em> that
already contains the SNP variables of class <em>snp</em>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="va">rs1422993</span>, data <span class="op">=</span> <span class="va">asthma.s</span><span class="op">)</span></span></code></pre></div>
<pre><code>
SNP: rs1422993  adjusted by: 
                0    %   1    %   OR lower upper  p-value  AIC
Codominant                                                    
G/G           730 59.0 173 50.9 1.00             0.017768 1642
G/T           425 34.3 145 42.6 1.44  1.12  1.85              
T/T            83  6.7  22  6.5 1.12  0.68  1.84              
Dominant                                                      
G/G           730 59.0 173 50.9 1.00             0.007826 1642
G/T-T/T       508 41.0 167 49.1 1.39  1.09  1.77              
Recessive                                                     
G/G-G/T      1155 93.3 318 93.5 1.00             0.877863 1649
T/T            83  6.7  22  6.5 0.96  0.59  1.57              
Overdominant                                                  
G/G-T/T       813 65.7 195 57.4 1.00             0.005026 1641
G/T           425 34.3 145 42.6 1.42  1.11  1.82              
log-Additive                                                  
0,1,2        1238 78.5 340 21.5 1.22  1.01  1.47 0.040151 1644</code></pre>
<p>The function <em>association</em> follows the usual syntax of R
modelling functions with the difference that the variables in the model
that are of class <em>snp</em> are tested using different genetic
models. In our example, we observe that all genetic models but the
recessive one are statistically significant. <em>association</em> also
fits the overdominant model, which compares the two homozygous genotypes
versus the heterozygous one. This genetic model of inheritance is
biologically rare although it has been linked to sickle cell anemia in
humans. The result table describes the number of individuals in each
genotype across cases and controls. The ORs and CI-95% are also
computed. The last column describes the AIC (Akaike information
criteria) that can be used to decide which is the best model of
inheritance; the lower the better the model is.</p>
<p>In the example, one may conclude that rs1422993 is associated with
asthma and that, for instance, the risk of being asthmatic is 39% higher
in people having at least one alternative allele (T) with respect to
individuals having none (dominant model). This risk is statistically
significant since the CI-95% does not contain 1, the P-value is
0.0078&lt;0.022, or the P-value of the max-statistics is 0.01</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/maxstat.html">maxstat</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">$</span><span class="va">casecontrol</span>, <span class="va">asthma.s</span><span class="op">$</span><span class="va">rs1422993</span><span class="op">)</span></span></code></pre></div>
<pre><code>     dominant recessive log-additive MAX-statistic Pr(&gt;z)  
[1,]    7.073     0.024        4.291         7.073 0.0143 *
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<p>If an expected model of inheritance is hypothesized, the association
analysis for the model can be specified in the argument <em>model</em>,
which by default test all models,</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="va">rs1422993</span>, <span class="va">asthma.s</span>, model<span class="op">=</span><span class="st">"dominant"</span><span class="op">)</span></span></code></pre></div>
<pre><code>
SNP: rs1422993  adjusted by: 
           0  %   1    %   OR lower upper  p-value  AIC
Dominant                                               
G/G      730 59 173 50.9 1.00             0.007826 1642
G/T-T/T  508 41 167 49.1 1.39  1.09  1.77              </code></pre>
<p>Association tests are typically adjusted by covariates, which are
incorporated in the model in the usual form</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="va">rs1422993</span> <span class="op">+</span> <span class="va">country</span> <span class="op">+</span> <span class="va">smoke</span>, <span class="va">asthma.s</span><span class="op">)</span></span></code></pre></div>
<pre><code>
SNP: rs1422993  adjusted by: country smoke 
                0    %   1    %   OR lower upper p-value  AIC
Codominant                                                   
G/G           728 59.1 173 51.0 1.00             0.06940 1408
G/T           423 34.3 144 42.5 1.38  1.05  1.82             
T/T            81  6.6  22  6.5 1.06  0.61  1.85             
Dominant                                                     
G/G           728 59.1 173 51.0 1.00             0.03429 1407
G/T-T/T       504 40.9 166 49.0 1.33  1.02  1.73             
Recessive                                                    
G/G-G/T      1151 93.4 317 93.5 1.00             0.79338 1411
T/T            81  6.6  22  6.5 0.93  0.54  1.60             
Overdominant                                                 
G/G-T/T       809 65.7 195 57.5 1.00             0.02147 1406
G/T           423 34.3 144 42.5 1.37  1.05  1.80             
log-Additive                                                 
0,1,2        1232 78.4 339 21.6 1.19  0.96  1.46 0.11191 1409</code></pre>
<p>ORs for stratified analysis on given categorical covariates are used
to verify whether the risk is constant across groups</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="va">rs1422993</span> <span class="op">+</span> <span class="fu">survival</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/strata.html" class="external-link">strata</a></span><span class="op">(</span><span class="va">gender</span><span class="op">)</span>, <span class="va">asthma.s</span><span class="op">)</span></span></code></pre></div>
<pre><code>
SNP: rs1422993  adjusted by: survival::strata(gender) 
                0    %   1    %   OR lower upper  p-value  AIC
Codominant                                                    
G/G           730 59.0 173 50.9 1.00             0.022940 1634
G/T           425 34.3 145 42.6 1.42  1.11  1.83              
T/T            83  6.7  22  6.5 1.09  0.66  1.80              
Dominant                                                      
G/G           730 59.0 173 50.9 1.00             0.011144 1633
G/T-T/T       508 41.0 167 49.1 1.37  1.07  1.74              
Recessive                                                     
G/G-G/T      1155 93.3 318 93.5 1.00             0.805330 1640
T/T            83  6.7  22  6.5 0.94  0.58  1.53              
Overdominant                                                  
G/G-T/T       813 65.7 195 57.4 1.00             0.006378 1632
G/T           425 34.3 145 42.6 1.41  1.10  1.80              
log-Additive                                                  
0,1,2        1238 78.5 340 21.5 1.21  1.00  1.46 0.055231 1636</code></pre>
<p>We can see, for instance, that the dominant model is significant only
in males. The subset argument allows fitting the model in a subgroup of
individuals</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="va">rs1422993</span>, <span class="va">asthma.s</span>, </span>
<span>                subset<span class="op">=</span><span class="va">country</span><span class="op">==</span><span class="st">"Spain"</span><span class="op">)</span></span></code></pre></div>
<pre><code>
SNP: rs1422993  adjusted by: 
               0    %  1    %   OR lower upper p-value   AIC
Codominant                                                  
G/G          179 54.6 22 44.9 1.00              0.3550 295.2
G/T          125 38.1 24 49.0 1.56  0.84  2.91              
T/T           24  7.3  3  6.1 1.02  0.28  3.66              
Dominant                                                    
G/G          179 54.6 22 44.9 1.00              0.2059 293.7
G/T-T/T      149 45.4 27 55.1 1.47  0.81  2.70              
Recessive                                                   
G/G-G/T      304 92.7 46 93.9 1.00              0.7576 295.2
T/T           24  7.3  3  6.1 0.83  0.24  2.85              
Overdominant                                                
G/G-T/T      203 61.9 25 51.0 1.00              0.1502 293.2
G/T          125 38.1 24 49.0 1.56  0.85  2.85              
log-Additive                                                
0,1,2        328 87.0 49 13.0 1.23  0.77  1.96  0.3816 294.5</code></pre>
<p>These analyses can be also be performed in quantitative traits, such
as body mass index, since <em>association</em> function automatically
selects the error distribution of the regression analysis (either
Gaussian or binomial).</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">bmi</span> <span class="op">~</span> <span class="va">rs1422993</span>, <span class="va">asthma.s</span><span class="op">)</span> </span></code></pre></div>
<pre><code>
SNP: rs1422993  adjusted by: 
                n    me     se       dif   lower  upper p-value  AIC
Codominant                                                          
G/G           896 25.53 0.1446  0.000000                 0.9069 9069
G/T           565 25.50 0.1834 -0.027059 -0.4874 0.4332             
T/T           105 25.71 0.4676  0.178076 -0.7057 1.0619             
Dominant                                                            
G/G           896 25.53 0.1446  0.000000                 0.9818 9067
G/T-T/T       670 25.54 0.1710  0.005089 -0.4324 0.4426             
Recessive                                                           
G/G-G/T      1461 25.52 0.1135  0.000000                 0.6694 9067
T/T           105 25.71 0.4676  0.188540 -0.6769 1.0540             
Overdominant                                                        
G/G-T/T      1001 25.55 0.1383  0.000000                 0.8424 9067
G/T           565 25.50 0.1834 -0.045739 -0.4965 0.4050             
log-Additive                                                        
0,1,2                           0.033951 -0.3153 0.3832  0.8489 9067</code></pre>
<p>For BMI, <em>association</em> tests whether the difference between
means is statistically significant, rather than computing an OR.</p>
<p>For multiple SNP data, our objective is to identify the variants that
are significantly associated with the trait. The most basic strategy is,
therefore, to fit an association test like the one described above for
each of the SNPs in the dataset and determine which of those
associations are significant. The massive univariate testing is the most
widely used analysis method for <em>omic</em> data because of its
simplicity. In <em>SNPassoc</em>, this type of analysis is done with the
function <em>WGassociation</em></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/WGassociation.html">WGassociation</a></span><span class="op">(</span><span class="va">casecontrol</span>, data<span class="op">=</span><span class="va">asthma.s</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ans</span><span class="op">)</span></span></code></pre></div>
<pre><code>           comments codominant dominant recessive overdominant log-additive
rs4490198         -    0.52765  0.29503   0.96400      0.29998      0.49506
rs4849332         -    0.96912  0.92986   0.84806      0.82327      0.97049
rs1367179         -    0.62775  0.59205   0.35786      0.86419      0.43994
rs11123242        -    0.68622  0.67596   0.39801      0.92878      0.52009
rs13014858        -    0.52578  0.26739   0.88011      0.34966      0.40897
rs1430094         -    0.13375  0.10569   0.54432      0.04490      0.36611</code></pre>
<p>Here, only the outcome is required in the formula argument (first
argument) since the function successively calls <em>association</em> on
each of the variables of class <em>snp</em> within <em>data</em>. The
function returns the P-values of association of each SNP under each
genetic model. Covariates can also be introduced in the model</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/WGassociation.html">WGassociation</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="va">country</span> <span class="op">+</span> <span class="va">smoke</span>, <span class="va">asthma.s</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ans.adj</span><span class="op">)</span></span></code></pre></div>
<p><code>SNPassoc</code> is computationally limited on large genomic
data. The computing time can be reduced by parallelization, specifying
in the argument <em>mc.cores</em> the number of computing cores to be
used. Alternatively, the function <em>scanWGassociation</em>, a C
compiled function, can be used to compute a predetermined genetic model
across all SNPs, passed in the argument <em>model</em>, which by default
is the additive model</p>
<pre class="more"><code><span><span class="va">ans.fast</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scanWGassociation.html">scanWGassociation</a></span><span class="op">(</span><span class="va">casecontrol</span>, <span class="va">asthma.s</span><span class="op">)</span></span></code></pre>
<p><strong>NOTE</strong>: This function is not available on the
<code>SNPassoc</code> version available on CRAN. The user can install
the development version available on GitHub to get access to this
function just executing</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"isglobal-brge/SNPassoc"</span><span class="op">)</span></span></code></pre></div>
<p>The P-values obtained from massive univariate analyses are visualized
with the generic <em>plot</em> function</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ans</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="SNPassoc_files/figure-html/plotGWAS-1.png" alt="Manhattan-type plots for different genetic models. P-values in -log10 scale to assess the association between case-control status and SNPs in the asthma example." width="700"><p class="caption">
Manhattan-type plots for different genetic models. P-values in -log10
scale to assess the association between case-control status and SNPs in
the asthma example.
</p>
</div>
<p>This produces a Manhattan plot of the -log10(P-values) for all the
SNPs over all models. It shows the nominal level of significance and the
Bonferroni level, which is the level corrected by the multiple testing
across all SNPs. The overall hypothesis of massive univariate
association tests is whether there is any SNP that is significantly
associated with the phenotype. As multiple SNPs are tested, the
probability of finding at least one significant finding increases if we
do not lower the significance threshold. The Bonferroni correction
lowers the threshold by the number of SNPs tested (0.0001=0.05/51). In
the Manhattan plotof our analysis, we see that no SNP is significant at
the Bonferroni level, and therefore there is no SNP that is
significantly associated with asthma.</p>
<p>Maximum-statistic (see <span class="citation">González et al.
(2008)</span>) can also be used to test association between asthma
status and SNPs</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans.max</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/maxstat.html">maxstat</a></span><span class="op">(</span><span class="va">asthma.s</span>, <span class="va">casecontrol</span><span class="op">)</span></span>
<span><span class="va">ans.max</span></span></code></pre></div>
<pre><code>           dominant recessive log-additive MAX-statistic  Pr(&gt;z)   
rs4490198     1.097     0.002        0.466         1.097 0.50210   
rs4849332     0.008     0.037        0.001         0.037 0.97653   
rs1367179     0.287     0.845        0.602         0.845 0.58995   
rs11123242    0.175     0.714        0.417         0.714 0.63913   
rs13014858    1.230     0.023        0.683         1.230 0.46374   
rs1430094     2.617     0.368        0.821         2.617 0.20637   
rs1430093     1.051     0.042        0.743         1.051 0.51935   
rs746710      0.728     0.679        1.051         1.051 0.51552   
rs1430090     0.172     0.463        0.000         0.463 0.74251   
rs6737251     0.143     0.156        0.217         0.217 0.86927   
rs11685217    0.894     0.030        0.705         0.894 0.56868   
rs1430097     0.003     0.183        0.029         0.183 0.88817   
rs10496465    0.003     0.020        0.008         0.020 0.98746   
rs3756688     0.016     0.738        0.266         0.738 0.62552   
rs2303063     0.060     1.271        0.658         1.271 0.45352   
rs1422993     7.073     0.024        4.291         7.073 0.01435 * 
rs2400478     1.662     0.056        1.055         1.662 0.35861   
rs714588      0.659     0.061        0.150         0.659 0.65844   
rs1023555     0.221     0.104        0.261         0.261 0.84669   
rs898070      0.020     1.794        0.346         1.794 0.33051   
rs963218      0.165     0.190        0.263         0.263 0.84348   
rs1419835     1.084     0.775        0.295         1.084 0.51065   
rs765023      1.959     0.526        0.483         1.959 0.30221   
rs1345267     2.582     0.003        1.383         2.582 0.21190   
rs324381      0.175     0.402        0.377         0.402 0.77509   
rs184448      9.710     2.026        8.253         9.710 0.00303 **
rs324396      2.312     0.188        1.051         2.312 0.24695   
rs324957      7.598     2.134        7.157         7.598 0.01405 * 
rs324960      2.901     7.928        6.443         7.928 0.01135 * 
rs10486657    0.054     0.500        0.001         0.500 0.73011   
rs324981      3.168     2.431        4.270         4.270 0.08118 . 
rs1419780     0.123     1.193        0.005         1.193 0.47745   
rs325462      1.545     1.645        2.412         2.412 0.23122   
rs727162      3.022     0.643        3.074         3.074 0.16054   
rs10250709    0.737     0.007        0.360         0.737 0.62823   
rs6958905     0.490     0.099        0.447         0.490 0.73247   
rs10238983    0.094     0.692        0.328         0.692 0.64741   
rs4941643     1.294     0.006        0.465         1.294 0.44617   
rs3794381     0.127     0.998        0.489         0.998 0.53592   
rs2031532     0.025     0.240        0.127         0.240 0.85706   
rs2247119     0.379     0.000        0.229         0.379 0.78481   
rs8000149     0.020     0.043        0.043         0.043 0.97255   
rs2274276     0.005     0.034        0.023         0.034 0.97747   
rs7332573     1.247     1.703        1.825         1.825 0.32938   
rs3829366     1.137     0.448        0.069         1.137 0.49137   
rs6084432     2.999     0.848        3.279         3.279 0.14324   
rs512625      2.180     0.030        1.119         2.180 0.26537   
rs3918395     0.903     0.414        0.455         0.903 0.57180   
rs2787095     0.178     0.069        0.184         0.184 0.88696   
rs2853215     0.686     0.931        1.130         1.130 0.49224   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<p>We note that even under the max-statistics none of the SNPs tested is
significant under the Bonferroni correction (&lt;0.0001) for multiple
SNP testing</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#minimum P-value across SNPs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">ans.max</span><span class="op">[</span><span class="st">"Pr(&gt;z)"</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 0.003029431</code></pre>
<p>Information for specific association models for given SNPs can also
be retrieved with <em>WGstats</em></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">infoTable</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/WGassociation.html">WGstats</a></span><span class="op">(</span><span class="va">ans</span><span class="op">)</span></span></code></pre></div>
<p>Therefore, we can have access to the results for a given SNP by</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">infoTable</span><span class="op">$</span><span class="va">rs1422993</span></span></code></pre></div>
<pre><code>
SNP: rs1422993  adjusted by: 
                0    %   1    %   OR lower upper  p-value  AIC
Codominant                                                    
G/G           730 59.0 173 50.9 1.00             0.017768 1642
G/T           425 34.3 145 42.6 1.44  1.12  1.85              
T/T            83  6.7  22  6.5 1.12  0.68  1.84              
Dominant                                                      
G/G           730 59.0 173 50.9 1.00             0.007826 1642
G/T-T/T       508 41.0 167 49.1 1.39  1.09  1.77              
Recessive                                                     
G/G-G/T      1155 93.3 318 93.5 1.00             0.877863 1649
T/T            83  6.7  22  6.5 0.96  0.59  1.57              
Overdominant                                                  
G/G-T/T       813 65.7 195 57.4 1.00             0.005026 1641
G/T           425 34.3 145 42.6 1.42  1.11  1.82              
log-Additive                                                  
0,1,2        1238 78.5 340 21.5 1.22  1.01  1.47 0.040151 1644</code></pre>
<p>recovering our previous results given by <em>association</em>
function.</p>
<p><strong>NOTE</strong>: The R output of specific association analyses
can be exported into LaTeX by using <em>getNiceTable</em> function and
<em>xtable</em> R package. The following code creates a table for the
SNPs rs1422993 and rs184448</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://xtable.r-forge.r-project.org/" class="external-link">xtable</a></span><span class="op">)</span></span>
<span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getNiceTable.html">getNiceTable</a></span><span class="op">(</span><span class="va">ans</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs1422993"</span>, <span class="st">"rs184448"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">nlines</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">out</span>, <span class="st">"nlines"</span><span class="op">)</span></span>
<span><span class="va">hlines</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/cumsum.html" class="external-link">cumsum</a></span><span class="op">(</span><span class="va">nlines</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/xtable/man/xtable.html" class="external-link">xtable</a></span><span class="op">(</span><span class="va">out</span>, caption<span class="op">=</span><span class="st">'Genetic association using</span></span>
<span><span class="st">                different genetic models from asthma </span></span>
<span><span class="st">                data example of rs1422993 and rs184448 </span></span>
<span><span class="st">                SNPs obtained with SNPassoc.'</span>,</span>
<span>             label <span class="op">=</span> <span class="st">'tab-2SNPs'</span><span class="op">)</span>,</span>
<span>      tabular.enviroment<span class="op">=</span><span class="st">"longtable"</span>, file<span class="op">=</span><span class="st">"tableSNPs"</span>,</span>
<span>      floating<span class="op">=</span><span class="cn">FALSE</span>,  include.rownames <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>      hline.after<span class="op">=</span> <span class="va">hlines</span>, sanitize.text.function<span class="op">=</span><span class="va">identity</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="gene-environment-and-gene-gene-interactions">Gene-environment and gene-gene interactions<a class="anchor" aria-label="anchor" href="#gene-environment-and-gene-gene-interactions"></a>
</h2>
<p>Gene-enviroment (GxE) analyses can be performed within
<code>SNPassoc</code> using <em>association</em> function. Assume that
we are interested in testing whether the risk of rs1422993 for asthma
under the dominant model is different among smokers (variable smoke;
0=never, 1=ever). This code fits a model with an interaction term where
the environmental variable is required to be a factor factor
variable.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="fu"><a href="../reference/inheritance.html">dominant</a></span><span class="op">(</span><span class="va">rs1422993</span><span class="op">)</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">smoke</span><span class="op">)</span>, </span>
<span>            data<span class="op">=</span><span class="va">asthma.s</span><span class="op">)</span></span></code></pre></div>
<pre><code>
      SNP: dominant(rs1422993  adjusted by: 
 Interaction 
---------------------
          0       OR lower upper   1      OR lower upper
G/G     486 130 1.00    NA    NA 242 43 0.66  0.46  0.97
G/T-T/T 354 128 1.35  1.02  1.79 150 38 0.95  0.63  1.42

p interaction: 0.8513 

 factor(smoke) within dominant(rs1422993 
---------------------
G/G 
    0   1   OR lower upper
0 486 130 1.00    NA    NA
1 242  43 0.66  0.46  0.97

G/T-T/T 
    0   1  OR lower upper
0 354 128 1.0    NA    NA
1 150  38 0.7  0.47  1.06

p trend: 0.8513 

 dominant(rs1422993 within factor(smoke) 
---------------------
0 
          0   1   OR lower upper
G/G     486 130 1.00    NA    NA
G/T-T/T 354 128 1.35  1.02  1.79

1 
          0  1   OR lower upper
G/G     242 43 1.00    NA    NA
G/T-T/T 150 38 1.43  0.88  2.31

p trend: 0.8513 </code></pre>
<p>The result is an interaction table showing that the risk of
individuals carrying the T allele increases the risk of asthma in never
smokers (OR=1.35; CI: 1.02-1.79) while it is not significant in ever
smokers (OR=0.95; CI: 0.63-1.42). However, the interaction is not
statistically significant (<span class="math inline">\(P\)</span>-interaction=0.8513). The output also
shows the stratified ORs that can help in interpreting the results.</p>
<p>In a similar way, gene-gene interaction (GxG) of a given SNP
epistasis model can also be fitted using the same function. In that
case, the genetic model of the interacting SNP must be indicated in the
<em>model.inteaction</em> argument.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/association.html">association</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="va">rs1422993</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">rs184448</span><span class="op">)</span>, </span>
<span>            data<span class="op">=</span><span class="va">asthma.s</span>, model.interaction <span class="op">=</span> <span class="st">"dominant"</span> <span class="op">)</span></span></code></pre></div>
<pre><code>
      SNP: rs1422993  adjusted by: 
 Interaction 
---------------------
        T/T      OR lower upper T/G      OR lower upper   0  1  G/G lower upper
G/G     227 43 1.00    NA    NA 359 96 1.41  0.95  2.10 128 30 1.24  0.74  2.07
G/T-T/T 154 33 1.13  0.69  1.86 265 93 1.85  1.24  2.77  78 38 2.57  1.55  4.27

p interaction: 0.24499 

 factor(rs184448) within rs1422993 
---------------------
G/G 
      0  1   OR lower upper
T/T 227 43 1.00    NA    NA
T/G 359 96 1.41  0.95  2.10
G/G 128 30 1.24  0.74  2.07

G/T-T/T 
      0  1   OR lower upper
T/T 154 33 1.00    NA    NA
T/G 265 93 1.64  1.05  2.55
G/G  78 38 2.27  1.32  3.90

p trend: 0.24499 

 rs1422993 within factor(rs184448) 
---------------------
T/T 
          0  1   OR lower upper
G/G     227 43 1.00    NA    NA
G/T-T/T 154 33 1.13  0.69  1.86

T/G 
          0  1   OR lower upper
G/G     359 96 1.00    NA    NA
G/T-T/T 265 93 1.31  0.95  1.82

G/G 
          0  1   OR lower upper
G/G     128 30 1.00    NA    NA
G/T-T/T  78 38 2.08  1.19  3.62

p trend: 0.12743 </code></pre>
<p>We observe that the interaction between these two SNPs is not
statistically significant (P-value=0.24). However, the OR of GG genotype
of rs184448 differs across individuals between the GG and GT-TT
genotypes of rs1422993 (see ORs for GG in the second table of the
output).</p>
<p>The user also can perform GxG for a set of SNPs using this code. Let
us assume we are interested in assessing interaction between the SNPs
that are significant at 10% level</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/WGassociation.html">WGassociation</a></span><span class="op">(</span><span class="va">casecontrol</span>, data<span class="op">=</span><span class="va">asthma.s</span><span class="op">)</span></span>
<span><span class="va">mask</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">ans</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">x</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">&lt;</span><span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="va">sig.snps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mask</span><span class="op">[</span><span class="va">mask</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">sig.snps</span></span></code></pre></div>
<pre><code> [1] "rs1430094" "rs1422993" "rs765023"  "rs184448"  "rs324396"  "rs324957" 
 [7] "rs324960"  "rs324981"  "rs727162"  "rs6084432"</code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">)</span><span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span><span class="va">sig.snps</span><span class="op">)</span></span>
<span><span class="va">asthma.s2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setupSNP.html">setupSNP</a></span><span class="op">(</span><span class="va">asthma</span>, colSNPs <span class="op">=</span> <span class="va">idx</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span>
<span><span class="va">ans.int</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/interactionPval.html">interactionPval</a></span><span class="op">(</span><span class="va">casecontrol</span> <span class="op">~</span> <span class="fl">1</span>, data<span class="op">=</span><span class="va">asthma.s2</span><span class="op">)</span></span>
<span><span class="va">ans.int</span></span></code></pre></div>
<pre><code>            rs1430094   rs1422993    rs765023    rs184448    rs324396
rs1430094 0.132526514 0.653457029 0.816154586 0.787386835 0.694311497
rs1422993 0.140433948 0.016719949 0.133131712 0.375246182 0.683959376
rs765023  0.171993821 0.144029395 0.182805988 0.961405419 0.194520134
rs184448  0.100074594 0.019963426 0.034666219 0.007969948 0.036825613
rs324396  0.204742661 0.190583402 0.131945703 0.344952544 0.209589962
rs324957  0.103680541 0.019820456 0.034446832 0.696623215 0.318903026
rs324960  0.107216765 0.024180405 0.174801032 0.291457716 0.584132142
rs324981  0.128489057 0.144077239 0.153521356 0.646139555 0.301782199
rs727162  0.258828613 0.240900610 0.188971595 0.173341993 0.120240909
rs6084432 0.214413938 0.187372912 0.187830183 0.264160389 0.252965105
             rs324957    rs324960    rs324981    rs727162   rs6084432
rs1430094 0.644661690 0.096913557 0.774083175 0.043758067 0.946641185
rs1422993 0.257661509 0.175946789 0.095251421 0.577504655 0.027437424
rs765023  0.437535350 0.614095008 0.117705086 0.217580773 0.896707843
rs184448  0.018290318 0.356124918 0.311031078 0.297592380 0.349332875
rs324396  0.232076552 0.747574191 0.064231657 0.361336095 0.388626207
rs324957  0.019578585 0.595796247 0.965290164 0.402781904 0.272919867
rs324960  0.404830873 0.017401694 0.168077057 0.406220478 0.725774374
rs324981  0.657378128 0.533095852 0.117043376 0.463436271 0.862291944
rs727162  0.174960014 0.500811687 0.167364803 0.206131411 0.885294485
rs6084432 0.174006927 0.146808311 0.210647392 0.203957720 0.193711884</code></pre>
<p>we can visualize the results by</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ans.int</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="SNPassoc_files/figure-html/plotInf-1.png" alt="Interaction plot. Interaction plot of SNPs significant al 10\% significant level (see help of 'interactionPval'  function to see what is represented in the plot). " width="672"><p class="caption">
Interaction plot. Interaction plot of SNPs significant al 10%
significant level (see help of ‘interactionPval’ function to see what is
represented in the plot).
</p>
</div>
</div>
<div class="section level2">
<h2 id="haplotype-analysis">Haplotype analysis<a class="anchor" aria-label="anchor" href="#haplotype-analysis"></a>
</h2>
<p>Genetic association studies can be extended from single SNP
associations to haplotype associations. Several examples including
different complex diseases can be found in . While alleles naturally
occur in haplotypes, as they belong to chromosomes, the phase, or the
knowledge of the chromosome an allele belongs to is lost in the
genotyping process. As each genotype is measured with a different probe
the phase between the alleles in a chromosome is broken. Consider for
instance an individual for which one chromosome has alleles A and T at
two different loci and alleles G or C at the second chromosome. The
individual’s genotypes for the individual at the two loci are A/G and
T/C, which are the same genotypes of another individual that has alleles
A and C in one chromosome and G and T in the second chromosome. Clearly,
if the pair A-T confers a risk to a disease, subject one is at risk
while subject two is not, despite both of them having the same
genotypes.</p>
<p>The only unequivocal method of resolving phase ambiguity is
sequencing the chromosomes of individuals. However, given the
correlational structure of SNPs, it is possible to estimate the
probability of a particular haplotype in a subject. This can be done in
a genetic association study where a number of cases and controls are
genotyped. There are numerous methods to infer unobserved haplotypes,
two of the most popular are maximum likelihood, implemented via the
expectation-maximization (EM) algorithm , and a parsimony method .
Recent methods based on Bayesian models have also been proposed . In
addition, haplotypes inferences carry uncertainty, which should be
considered in association analyses.</p>
<div class="section level3">
<h3 id="haplotype-estimation">Haplotype estimation<a class="anchor" aria-label="anchor" href="#haplotype-estimation"></a>
</h3>
<p>We now illustrate how to perform haplotype estimation from genotype
data using the EM algorithm and how to integrate haplotype uncertainty
when evaluating the association between traits and haplotypes. Haplotype
inference is performed with <code>haplo.stats</code>, for which
genotypes are encoded in a different format. <em>make.geno</em>, from
<code>SNPassoc</code>, formats data for <code>haplo.stats</code>. The
function <em>haplo.em</em> computes the haplotype frequency in the data
for the SNPs of interest. Here we illustrate how to estimate haplotypes
built from the SNPs rs714588, rs1023555 and rs898070:</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://analytictools.mayo.edu/research/haplo-stats/" class="external-link">haplo.stats</a></span><span class="op">)</span></span>
<span><span class="va">snpsH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs714588"</span>, <span class="st">"rs1023555"</span>,  <span class="st">"rs898070"</span><span class="op">)</span></span>
<span><span class="va">genoH</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makegeno.html">make.geno</a></span><span class="op">(</span><span class="va">asthma.s</span>, <span class="va">snpsH</span><span class="op">)</span></span>
<span><span class="va">em</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.em.html" class="external-link">haplo.em</a></span><span class="op">(</span><span class="va">genoH</span>, locus.label <span class="op">=</span> <span class="va">snpsH</span>, miss.val <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">em</span></span></code></pre></div>
<pre><code>================================================================================ 
                                   Haplotypes                                    
================================================================================ 
  rs714588 rs1023555 rs898070 hap.freq
1        1         1        1  0.04090
2        1         1        2  0.02439
3        1         2        1  0.04265
4        1         2        2  0.44047
5        2         1        1  0.08271
6        2         1        2  0.08403
7        2         2        1  0.20794
8        2         2        2  0.07691
================================================================================ 
                                    Details                                      
================================================================================ 
lnlike =  -4102.691 
lr stat for no LD =  774.5411 , df =  4 , p-val =  0 </code></pre>
<p>Coding the common and variant alleles as 1 and 2, we can see there
are 8 possible haplotypes across the subjects and are listed with an
estimated haplotype frequency. Clearly, the haplotypes are not equally
probable, as expected from the high LD between the SNPs. In particular,
we observe that haplotypes 4 and 7 are the most probable, accumulating
65% of the haplotype sample. <em>haplo.em</em> estimates for each
subject the probability of a given haplotype in each of the subject’s
chromosomes.</p>
</div>
<div class="section level3">
<h3 id="haplotype-association">Haplotype association<a class="anchor" aria-label="anchor" href="#haplotype-association"></a>
</h3>
<p>We then want to assess if any of these haplotypes significantly
associates with asthma. The <em>haplo.glm</em> fits a regression model
between the phenotype and the haplotypes, incorporating the uncertainty
for the probable haplotypes of individuals. The function
<em>intervals</em> of <code>SNPassoc</code> provides a nice summary of
the results</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trait</span> <span class="op">&lt;-</span> <span class="va">asthma.s</span><span class="op">$</span><span class="va">casecontrol</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.html" class="external-link">haplo.glm</a></span><span class="op">(</span><span class="va">trait</span> <span class="op">~</span> <span class="va">genoH</span>,           </span>
<span>                 family<span class="op">=</span><span class="st">"binomial"</span>, </span>
<span>                 locus.label<span class="op">=</span><span class="va">snpsH</span>,</span>
<span>                 allele.lev<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">genoH</span><span class="op">)</span><span class="op">$</span><span class="va">unique.alleles</span>,</span>
<span>                 control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.control.html" class="external-link">haplo.glm.control</a></span><span class="op">(</span>haplo.freq.min<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>   </span>
<span><span class="fu"><a href="../reference/intervals.html">intervals</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span></code></pre></div>
<pre><code>               freq   or   95%   C.I.   P-val 
         ATG 0.4405   1.00 Reference haplotype 
         GAA 0.0827   1.09 (   0.77 -   1.56 )  0.6160 
         GAG 0.0841   0.99 (   0.69 -   1.42 )  0.9642 
         GTA 0.2080   1.06 (   0.84 -   1.34 )  0.6379 
         GTG 0.0769   1.09 (   0.76 -   1.58 )  0.6367 
  genoH.rare 0.1079   1.15 (   0.83 -   1.59 )  0.3968 </code></pre>
<p><em>haplo.glm</em> fits a logistic regression model for the asthma
status (<em>trait</em> - NOTE: this must be a 0/1 variable) on the
inferred haplotypes (<em>genoH</em>), the names of SNPs and their allele
names are passed in the <em>locus.label</em> and <em>allele.lev</em>
arguments, while only haplotypes with at least 5% frequency are
considered. As a result, we obtain the OR for each haplotype with their
significance P-value, with respect to the most common haplotype (ATG
with 44% frequency). In particular, from this analysis, we cannot see
any haplotype significantly associated with asthma.</p>
</div>
<div class="section level3">
<h3 id="sliding-window-approach">Sliding window approach<a class="anchor" aria-label="anchor" href="#sliding-window-approach"></a>
</h3>
<p>The inference of haplotypes depends on a predefined region or sets of
SNPs. For instance, in the last section, we selected three SNPs that
were in high LD. However, when no previous knowledge is available about
the region or SNPs for which haplotypes should be inferred, we can apply
a sliding window for haplotype inference .</p>
<p>To illustrate this type of analysis, we now consider a second block
of 10 SNPs in our asthma example (from 6th to 15th SNP). Considering
large haplotypes, however, increases the number of possible haplotypes
in the sample, decreasing the power of finding real associations. In
addition, in predefined blocks, it is possible to miss the most
efficient length of the susceptible haplotype, incrementing the loss of
power. This is overcome using a sliding window. We thus ask which is the
haplotype combination from any of 4, 5, 6 or 7 consecutive SNPs that
gives the highest association with asthma status. We then reformat SNP
genotypes in the region with the <em>make.geno</em> function and perform
an association analysis for multiple haplotypes of <em>i</em> SNPs
sliding from the 6th to the 15th SNP in the data. We perform an analysis
for each window length <em>i</em> varying from 4 to 7 SNPs.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snpsH2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/labels.html" class="external-link">labels</a></span><span class="op">(</span><span class="va">asthma.s</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span></span>
<span><span class="va">genoH2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makegeno.html">make.geno</a></span><span class="op">(</span><span class="va">asthma.s</span>, <span class="va">snpsH2</span><span class="op">)</span></span>
<span><span class="va">haplo.score</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">4</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span> <span class="op">{</span></span>
<span> <span class="va">trait</span> <span class="op">&lt;-</span> <span class="va">asthma.s</span><span class="op">$</span><span class="va">casecontrol</span></span>
<span> <span class="va">haplo.score</span><span class="op">[[</span><span class="va">i</span><span class="op">-</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.score.slide.html" class="external-link">haplo.score.slide</a></span><span class="op">(</span><span class="va">trait</span>, <span class="va">genoH2</span>, </span>
<span>                          trait.type<span class="op">=</span><span class="st">"binomial"</span>,</span>
<span>                          n.slide<span class="op">=</span><span class="va">i</span>,</span>
<span>                          simulate<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                          sim.control<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/score.sim.control.html" class="external-link">score.sim.control</a></span><span class="op">(</span>min.sim<span class="op">=</span><span class="fl">100</span>,</span>
<span>                                       max.sim<span class="op">=</span><span class="fl">200</span><span class="op">)</span><span class="op">)</span> </span>
<span> <span class="op">}</span></span></code></pre></div>
<p>The results can be visualized with the following plot</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">4</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">haplo.score</span><span class="op">[[</span><span class="va">i</span><span class="op">-</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Sliding Window="</span>, <span class="va">i</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span> <span class="op">}</span></span></code></pre></div>
<div class="figure">
<img src="SNPassoc_files/figure-html/plotSliding-1.png" alt="Sliding window approach. Results obtained of varying haplotype size from 4 up to 7 of  6th to the 15th SNP from asthma data example" width="672"><p class="caption">
Sliding window approach. Results obtained of varying haplotype size from
4 up to 7 of 6th to the 15th SNP from asthma data example
</p>
</div>
<p>We observe that the highest -log10(P-value) is obtained for a
haplotype of 4 SNP length starting at the 4th SNP of the selected SNPs.
After deciding the best combination of SNPs, the haplotype association
with asthma can be estimated by</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snpsH3</span> <span class="op">&lt;-</span> <span class="va">snpsH2</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span>
<span><span class="va">genoH3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makegeno.html">make.geno</a></span><span class="op">(</span><span class="va">asthma.s</span>, <span class="va">snpsH3</span><span class="op">)</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.html" class="external-link">haplo.glm</a></span><span class="op">(</span><span class="va">trait</span><span class="op">~</span><span class="va">genoH3</span>,           </span>
<span>                 family<span class="op">=</span><span class="st">"binomial"</span>, </span>
<span>                 locus.label<span class="op">=</span><span class="va">snpsH3</span>,</span>
<span>                 allele.lev<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">genoH3</span><span class="op">)</span><span class="op">$</span><span class="va">unique.alleles</span>,</span>
<span>                 control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.control.html" class="external-link">haplo.glm.control</a></span><span class="op">(</span>haplo.freq.min<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>      </span>
<span><span class="fu"><a href="../reference/intervals.html">intervals</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span></code></pre></div>
<pre><code>               freq   or   95%   C.I.   P-val 
        TCCC 0.3521   1.00 Reference haplotype 
        GCCC 0.2963   1.01 (   0.82 -   1.24 )  0.9495 
        TTCA 0.1184   0.89 (   0.67 -   1.20 )  0.4433 
        TTTA 0.1830   1.12 (   0.87 -   1.43 )  0.3796 
 genoH3.rare 0.0502   0.78 (   0.50 -   1.22 )  0.2750 </code></pre>
<p>Here, we observe that individuals carrying the haplotype GCAC have a
53% increased risk of asthma relative to those having the reference
haplotype TCGT (OR=1.53, p=0.0021). The haplotype GTCA is also
significantly associated with the disease (p=0.0379). A likelihood ratio
test for haplotype status can be extracted from the results of the
<em>haplo.glme</em> function:</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lrt</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="va">lrt</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html" class="external-link">pchisq</a></span><span class="op">(</span><span class="va">lrt</span><span class="op">$</span><span class="va">lrt</span>, <span class="va">lrt</span><span class="op">$</span><span class="va">df</span>, lower<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 0.5044641</code></pre>
<p>We can also test the association between asthma and the haplotype
adjusted for smoking status</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">smoke</span> <span class="op">&lt;-</span> <span class="va">asthma.s</span><span class="op">$</span><span class="va">smoke</span></span>
<span><span class="va">mod.adj.ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">trait</span> <span class="op">~</span> <span class="va">smoke</span>, family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="va">mod.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.html" class="external-link">haplo.glm</a></span><span class="op">(</span><span class="va">trait</span> <span class="op">~</span> <span class="va">genoH3</span> <span class="op">+</span> <span class="va">smoke</span> ,           </span>
<span>                 family<span class="op">=</span><span class="st">"binomial"</span>, </span>
<span>                 locus.label<span class="op">=</span><span class="va">snpsH3</span>,</span>
<span>                 allele.lev<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">genoH3</span><span class="op">)</span><span class="op">$</span><span class="va">unique.alleles</span>,</span>
<span>                 control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.control.html" class="external-link">haplo.glm.control</a></span><span class="op">(</span>haplo.freq.min<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lrt.adj</span> <span class="op">&lt;-</span> <span class="va">mod.adj.ref</span><span class="op">$</span><span class="va">deviance</span> <span class="op">-</span> <span class="va">mod.adj</span><span class="op">$</span><span class="va">deviance</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html" class="external-link">pchisq</a></span><span class="op">(</span><span class="va">lrt.adj</span>, <span class="va">mod.adj</span><span class="op">$</span><span class="va">lrt</span><span class="op">$</span><span class="va">df</span>, lower<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 0.6457956</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.utf8  LC_CTYPE=Spanish_Spain.utf8   
[3] LC_MONETARY=Spanish_Spain.utf8 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] haplo.stats_1.8.9 arsenal_3.6.3     SNPassoc_2.0-17   BiocStyle_2.24.0 

loaded via a namespace (and not attached):
  [1] backports_1.4.1                        
  [2] Hmisc_4.7-1                            
  [3] BiocFileCache_2.4.0                    
  [4] systemfonts_1.0.4                      
  [5] plyr_1.8.7                             
  [6] splines_4.2.1                          
  [7] BiocParallel_1.30.4                    
  [8] GenomeInfoDb_1.32.4                    
  [9] ggplot2_3.3.6                          
 [10] TH.data_1.1-1                          
 [11] digest_0.6.29                          
 [12] htmltools_0.5.3                        
 [13] fansi_1.0.3                            
 [14] magrittr_2.0.3                         
 [15] checkmate_2.1.0                        
 [16] memoise_2.0.1                          
 [17] BSgenome_1.64.0                        
 [18] cluster_2.1.4                          
 [19] Biostrings_2.64.1                      
 [20] matrixStats_0.62.0                     
 [21] sandwich_3.0-2                         
 [22] pkgdown_2.0.6                          
 [23] prettyunits_1.1.1                      
 [24] jpeg_0.1-9                             
 [25] colorspace_2.0-3                       
 [26] blob_1.2.3                             
 [27] rappdirs_0.3.3                         
 [28] textshaping_0.3.6                      
 [29] xfun_0.33                              
 [30] dplyr_1.0.10                           
 [31] crayon_1.5.2                           
 [32] RCurl_1.98-1.9                         
 [33] jsonlite_1.8.2                         
 [34] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [35] survival_3.4-0                         
 [36] VariantAnnotation_1.42.1               
 [37] zoo_1.8-11                             
 [38] glue_1.6.2                             
 [39] poisbinom_1.0.1                        
 [40] gtable_0.3.1                           
 [41] zlibbioc_1.42.0                        
 [42] XVector_0.36.0                         
 [43] MatrixModels_0.5-1                     
 [44] DelayedArray_0.22.0                    
 [45] rms_6.3-0                              
 [46] BiocGenerics_0.42.0                    
 [47] SparseM_1.81                           
 [48] scales_1.2.1                           
 [49] mvtnorm_1.1-3                          
 [50] DBI_1.1.3                              
 [51] Rcpp_1.0.9                             
 [52] progress_1.2.2                         
 [53] htmlTable_2.4.1                        
 [54] foreign_0.8-83                         
 [55] bit_4.0.4                              
 [56] Formula_1.2-4                          
 [57] stats4_4.2.1                           
 [58] htmlwidgets_1.5.4                      
 [59] httr_1.4.4                             
 [60] RColorBrewer_1.1-3                     
 [61] ellipsis_0.3.2                         
 [62] farver_2.1.1                           
 [63] pkgconfig_2.0.3                        
 [64] XML_3.99-0.11                          
 [65] nnet_7.3-18                            
 [66] sass_0.4.2                             
 [67] dbplyr_2.2.1                           
 [68] deldir_1.0-6                           
 [69] utf8_1.2.2                             
 [70] labeling_0.4.2                         
 [71] tidyselect_1.2.0                       
 [72] rlang_1.0.6                            
 [73] AnnotationDbi_1.58.0                   
 [74] munsell_0.5.0                          
 [75] tools_4.2.1                            
 [76] cachem_1.0.6                           
 [77] cli_3.4.1                              
 [78] generics_0.1.3                         
 [79] RSQLite_2.2.18                         
 [80] evaluate_0.17                          
 [81] stringr_1.4.1                          
 [82] fastmap_1.1.0                          
 [83] yaml_2.3.5                             
 [84] ragg_1.2.3                             
 [85] org.Hs.eg.db_3.15.0                    
 [86] knitr_1.40                             
 [87] bit64_4.0.5                            
 [88] fs_1.5.2                               
 [89] purrr_0.3.5                            
 [90] KEGGREST_1.36.3                        
 [91] nlme_3.1-160                           
 [92] quantreg_5.94                          
 [93] xml2_1.3.3                             
 [94] biomaRt_2.52.0                         
 [95] compiler_4.2.1                         
 [96] rstudioapi_0.14                        
 [97] filelock_1.0.2                         
 [98] curl_4.3.3                             
 [99] png_0.1-7                              
[100] tibble_3.1.8                           
[101] bslib_0.4.0                            
[102] stringi_1.7.8                          
[103] highr_0.9                              
[104] GenomicFeatures_1.48.4                 
[105] desc_1.4.2                             
[106] lattice_0.20-45                        
[107] Matrix_1.5-1                           
[108] vctrs_0.4.2                            
[109] pillar_1.8.1                           
[110] lifecycle_1.0.3                        
[111] BiocManager_1.30.18                    
[112] jquerylib_0.1.4                        
[113] data.table_1.14.4                      
[114] bitops_1.0-7                           
[115] rtracklayer_1.56.1                     
[116] GenomicRanges_1.48.0                   
[117] R6_2.5.1                               
[118] BiocIO_1.6.0                           
[119] latticeExtra_0.6-30                    
[120] bookdown_0.29                          
[121] gridExtra_2.3                          
[122] IRanges_2.30.1                         
[123] codetools_0.2-18                       
[124] polspline_1.1.20                       
[125] MASS_7.3-58.1                          
[126] assertthat_0.2.1                       
[127] SummarizedExperiment_1.26.1            
[128] rjson_0.2.21                           
[129] rprojroot_2.0.3                        
[130] withr_2.5.0                            
[131] GenomicAlignments_1.32.1               
[132] Rsamtools_2.12.0                       
[133] multcomp_1.4-20                        
[134] S4Vectors_0.34.0                       
[135] GenomeInfoDbData_1.2.8                 
[136] parallel_4.2.1                         
[137] hms_1.1.2                              
[138] grid_4.2.1                             
[139] rpart_4.1.16                           
[140] tidyr_1.2.1                            
[141] rmarkdown_2.17                         
[142] MatrixGenerics_1.8.1                   
[143] Biobase_2.56.0                         
[144] base64enc_0.1-3                        
[145] restfulr_0.0.15                        
[146] interp_1.1-3                           </code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-gonzalez2007snpassoc" class="csl-entry">
González, Juan R, Lluı́s Armengol, Xavier Solé, Elisabet Guinó, Josep M
Mercader, Xavier Estivill, and Vı́ctor Moreno. 2007.
<span>“<span>SNP</span>assoc: <span>A</span>n <span>R</span> Package to
Perform Whole Genome Association Studies.”</span>
<em>Bioinformatics</em> 23 (5): 654–55.
</div>
<div id="ref-gonzalez2008maximizing" class="csl-entry">
González, Juan R, Josep L Carrasco, Frank Dudbridge, Lluı́s Armengol,
Xavier Estivill, and Victor Moreno. 2008. <span>“Maximizing Association
Statistics over Genetic Models.”</span> <em>Genetic Epidemiology</em> 32
(3): 246–54.
</div>
<div id="ref-wigginton2005note" class="csl-entry">
Wigginton, Janis E, David J Cutler, and Gonçalo R Abecasis. 2005.
<span>“A Note on Exact Tests of <span>H</span>ardy-<span>W</span>einberg
Equilibrium.”</span> <em>The American Journal of Human Genetics</em> 76
(5): 887–93.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Victor Moreno, Juan R Gonzalez, Dolors Pelegri.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
