<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function carries out an association analysis between a single SNP and
  a dependent variable (phenotype) under five different genetic models (inheritance patterns): 
  codominant, dominant, recessive, overdominant and log-additive. The phenotype may be quantitative 
  or categorical. In the second case (e.g. case-control studies) this variable must be of class 'factor' 
  with two levels."><title>Association analysis between a single SNP and a given phenotype — association • SNPassoc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Association analysis between a single SNP and a given phenotype — association"><meta property="og:description" content="This function carries out an association analysis between a single SNP and
  a dependent variable (phenotype) under five different genetic models (inheritance patterns): 
  codominant, dominant, recessive, overdominant and log-additive. The phenotype may be quantitative 
  or categorical. In the second case (e.g. case-control studies) this variable must be of class 'factor' 
  with two levels."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SNPassoc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0-17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/SNPassoc.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/brgelab" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/isglobal-brge/SNPassoc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Association analysis between a single SNP and a given phenotype</h1>
      
      <div class="d-none name"><code>association.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function carries out an association analysis between a single SNP and
  a dependent variable (phenotype) under five different genetic models (inheritance patterns): 
  codominant, dominant, recessive, overdominant and log-additive. The phenotype may be quantitative 
  or categorical. In the second case (e.g. case-control studies) this variable must be of class 'factor' 
  with two levels.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">association</span><span class="op">(</span><span class="va">formula</span>, <span class="va">data</span>, model<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span><span class="op">)</span>, model.interaction<span class="op">=</span></span>
<span>       <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"codominant"</span><span class="op">)</span>, <span class="va">subset</span>, name.snp <span class="op">=</span> <span class="cn">NULL</span>, quantitative <span class="op">=</span> </span>
<span>       <span class="fu"><a href="SNPassoc-internal.html">is.quantitative</a></span><span class="op">(</span><span class="va">formula</span>,<span class="va">data</span><span class="op">)</span>, genotypingRate<span class="op">=</span> <span class="fl">0</span>, </span>
<span>       level <span class="op">=</span> <span class="fl">0.95</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>formula</dt>
<dd><p>a symbolic description of the model to be fited (a formula object). 
                It might have either a continuous variable (quantitative traits) or a 
                factor variable (case-control studies) as the response on the left of the <code>~</code> 
                operator and a term corresponding to the SNP on the right. This term must be
                of class <code>snp</code> (e.g. <code>~</code>snp(var), where var is a given SNP), and it is 
                required. Terms with additional covariates on the right of the <code>~</code> operator may be 
                added to fit an adjusted model (e.g., <code>~</code>var1+var2+...+varN+SNP). The formula 
                allows to incorporate more than one object of class <code>snp</code>. In that case, 
                the analysis is done for the first SNP which appears in the formula adjusted by the others 
                covariates and other additional SNPs.</p></dd>

  <dt>data</dt>
<dd><p>a required dataframe of class 'setupSNP' containing the variables in the model.</p></dd>

  <dt>model</dt>
<dd><p>a character string specifying the type of genetic model (mode of inheritance) for the SNP. 
               This indicates how the genotypes should be collapsed. Possible
               values are "codominant", "dominant", "recessive", "overdominant", "additive" or "all". The default
               is "all" that fits the 5 possible genetic models. Only the first words are required, e.g "co", "do", etc.</p></dd>
              
  <dt>model.interaction</dt>
<dd><p>a character string specifying the type of genetic model (mode of inheritance) assumed for
                           the SNP when it is included in a interaction term. Possible values are "codominant", 
                           "dominant", "recessive", "overdominant". The default is "codominant".</p></dd>

  <dt>subset</dt>
<dd><p>an optional vector specifying a subset of observations to be used in the fitting process</p></dd>
              
  <dt>name.snp</dt>
<dd><p>optional label of the SNP variable to be printed.</p></dd>

  <dt>quantitative</dt>
<dd><p>logical value indicating whether the phenotype (that which is in the
                      left of the operator ~ in 'formula' argument) is quantitative. The function 
                      'is.quantitative' returns FALSE when the phenotype
                      is a variable with two categories (i.e. indicating case-control status). Thus,
                      it is not a required argument but it may be modified by the user.</p></dd>

  <dt>genotypingRate</dt>
<dd><p>minimum percentage of genotype rate for the SNP to be analyzed.
                  Default is 0% (e.g. all SNPs are analyzed). This parameter should not be
                  changed. It is used in the function 'WGassociation'.</p></dd>

  <dt>level</dt>
<dd><p>signification level for confidence intervals.</p></dd>

  <dt>...</dt>
<dd><p>Other arguments to be passed through glm function</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function should be called by the user when we are interested in analyzing an unique SNP.
  It is recommended to use <code><a href="WGassociation.html">WGassociation</a></code> function when more than one SNP is studied. <br></p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>For each genetic model (codominant, dominant, recessive, overdominant, and log-additive) the function gives 
  a matrix with sample size and percentages for each genotype, the Odds Ratio and its 95% confidence interval
  (taking the most frequent homozygous genotype as the reference), the p-value corresponding to the likelihood ratio test obtained 
  from a comparison with the null model, and the Akaike Information Criterion (AIC) of each genetic model. In the case 
  of analyzing a quantitative trait, the function returns a matrix with sample size, mean and standard errors for each genotype, 
  mean difference and its 95% confidence interval with respect to the most frequent homozygous genotype, 
  the p-value obtained from an overall gene effect and the Akaike Information Criterion (AIC) of each genetic model.</p>
  

<p>When an interaction term (a categorical covariate with an SNP) is included in the model, three different tables are given. 
  The first one correponds to the full interaction matrix where the ORs (or mean differences if a quantitative trait is analyzed) 
  are expressed with respect to the non variant genotype and the first category of the covariate. The other two tables show the ORs 
  and their 95% confidence intervals for both marginal models. P values for interaction and trend are also showed in the output.</p>
  

    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.</p>
<p>Iniesta R, Guino E, Moreno V. Statistical analysis of genetic polymorphisms in epidemiological studies.
Gac Sanit. 2005;19(4):333-41.</p>
<p>Elston RC. Introduction and overview. Statistical methods in genetic epidemiology. Stat Methods Med Res.
2000;9:527-41.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="WGassociation.html">WGassociation</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SNPs</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># first, we create an object of class 'setupSNP'</span></span></span>
<span class="r-in"><span><span class="va">datSNP</span><span class="op">&lt;-</span><span class="fu"><a href="setupSNP.html">setupSNP</a></span><span class="op">(</span><span class="va">SNPs</span>,<span class="fl">6</span><span class="op">:</span><span class="fl">40</span>,sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># case-control study, crude analysis</span></span></span>
<span class="r-in"><span><span class="fu">association</span><span class="op">(</span><span class="va">casco</span><span class="op">~</span><span class="va">snp10001</span>, data<span class="op">=</span><span class="va">datSNP</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP: snp10001  adjusted by: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               0    %   1    %   OR lower upper p-value   AIC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Codominant                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T          24 51.1  68 61.8 1.00             0.13234 193.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T          21 44.7  32 29.1 0.54  0.26  1.11              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           2  4.3  10  9.1 1.76  0.36  8.64              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dominant                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T          24 51.1  68 61.8 1.00             0.21184 194.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T-C/C      23 48.9  42 38.2 0.64  0.32  1.28              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Recessive                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/T      45 95.7 100 90.9 1.00             0.27152 194.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           2  4.3  10  9.1 2.25  0.47 10.69              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Overdominant                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/C      26 55.3  78 70.9 1.00             0.06129 192.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T          21 44.7  32 29.1 0.51  0.25  1.03              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> log-Additive                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0,1,2        47 29.9 110 70.1 0.87  0.51  1.47 0.59448 195.4</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># case-control study, adjusted by sex and arterial blood pressure</span></span></span>
<span class="r-in"><span><span class="fu">association</span><span class="op">(</span><span class="va">casco</span><span class="op">~</span><span class="va">sex</span><span class="op">+</span><span class="va">snp10001</span><span class="op">+</span><span class="va">blood.pre</span>, data<span class="op">=</span><span class="va">datSNP</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP: snp10001  adjusted by: sex blood.pre </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               0    %   1    %   OR lower upper p-value   AIC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Codominant                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T          24 51.1  68 61.8 1.00             0.15410 195.8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T          21 44.7  32 29.1 0.55  0.26  1.14              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           2  4.3  10  9.1 1.74  0.35  8.63              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dominant                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T          24 51.1  68 61.8 1.00             0.22859 196.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T-C/C      23 48.9  42 38.2 0.65  0.32  1.31              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Recessive                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/T      45 95.7 100 90.9 1.00             0.28494 196.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           2  4.3  10  9.1 2.22  0.46 10.70              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Overdominant                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/C      26 55.3  78 70.9 1.00             0.07188 194.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T          21 44.7  32 29.1 0.52  0.25  1.06              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> log-Additive                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0,1,2        47 29.9 110 70.1 0.87  0.51  1.49 0.60861 197.3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># quantitative trait, crude analysis</span></span></span>
<span class="r-in"><span><span class="fu">association</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span><span class="op">~</span><span class="va">snp10001</span>,data<span class="op">=</span><span class="va">datSNP</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP: snp10001  adjusted by: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                n     me      se     dif   lower    upper   p-value   AIC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Codominant                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T           92 10.637 0.05630  0.0000                  0.0005068 316.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T           53 10.351 0.10612 -0.2858 -0.5061 -0.06549                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           12  9.929 0.24389 -0.7073 -1.0994 -0.31526                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dominant                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T           92 10.637 0.05630  0.0000                  0.0008307 318.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T-C/C       65 10.273 0.09882 -0.3636 -0.5727 -0.15456                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Recessive                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/T      145 10.532 0.05375  0.0000                  0.0028978 320.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           12  9.929 0.24389 -0.6029 -0.9933 -0.21246                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Overdominant                                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/C      104 10.555 0.06087  0.0000                  0.0752824 326.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T           53 10.351 0.10612 -0.2042 -0.4276  0.01929                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> log-Additive                                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0,1,2                           -0.3247 -0.4849 -0.16442 0.0001093 314.4</span>
<span class="r-in"><span><span class="co"># quantitative trait, adjusted by sex</span></span></span>
<span class="r-in"><span><span class="fu">association</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span><span class="op">~</span><span class="va">snp10001</span><span class="op">+</span><span class="va">sex</span>,data<span class="op">=</span><span class="va">datSNP</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 3 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 3 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 3 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 3 after EncodeVars() -- should no longer happen!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SNP: snp10001  adjusted by: sex </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                n     me      se     dif   lower    upper   p-value   AIC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Codominant                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T           92 10.637 0.05630  0.0000                  2.772e-04 316.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T           53 10.351 0.10612 -0.2970 -0.5171 -0.07694                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           12  9.929 0.24389 -0.7422 -1.1358 -0.34861                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dominant                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T           92 10.637 0.05630  0.0000                  5.555e-04 318.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T-C/C       65 10.273 0.09882 -0.3774 -0.5871 -0.16762                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Recessive                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/T      145 10.532 0.05375  0.0000                  1.976e-03 321.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C           12  9.929 0.24389 -0.6297 -1.0218 -0.23761                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Overdominant                                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T-C/C      104 10.555 0.06087  0.0000                  6.889e-02 327.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T           53 10.351 0.10612 -0.2091 -0.4329  0.01461                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> log-Additive                                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0,1,2                           -0.3393 -0.5003 -0.17829 5.918e-05 314.3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span><span class="co"># Interaction analysis</span></span></span>
<span class="r-in"><span><span class="co">#</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Interaction SNP and factor</span></span></span>
<span class="r-in"><span><span class="fu">association</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span><span class="op">~</span><span class="va">snp10001</span><span class="op">*</span><span class="va">sex</span><span class="op">+</span><span class="va">blood.pre</span>, data<span class="op">=</span><span class="va">datSNP</span>, </span></span>
<span class="r-in"><span>          model<span class="op">=</span><span class="st">"codominant"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 3 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       SNP: snp10001  adjusted by: blood.pre </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Interaction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Male             dif   lower   upper    Female              dif   lower</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 40 10.65 0.08021  0.0000      NA      NA 52 10.626 0.07883 -0.02559 -0.2941</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 27 10.52 0.10292 -0.1298 -0.4481  0.1886 26 10.176 0.18415 -0.47199 -0.7943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C  8 10.01 0.34582 -0.6371 -1.1316 -0.1426  4  9.761 0.28635 -0.88728 -1.5571</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T  0.2429</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T -0.1497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C -0.2174</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p interaction: 0.36299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sex within snp10001 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         n    me      se      dif   lower  upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Male   40 10.65 0.08021  0.00000      NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Female 52 10.63 0.07883 -0.02559 -0.2941 0.2429</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         n    me     se     dif   lower   upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Male   27 10.52 0.1029  0.0000      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Female 26 10.18 0.1841 -0.3422 -0.6931 0.00864</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        n     me     se     dif  lower  upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Male   8 10.014 0.3458  0.0000     NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Female 4  9.761 0.2863 -0.2502 -1.032 0.5319</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p trend: 0.26751 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  snp10001 within sex </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Male </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n    me      se     dif   lower   upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 40 10.65 0.08021  0.0000      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 27 10.52 0.10292 -0.1298 -0.4481  0.1886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C  8 10.01 0.34582 -0.6371 -1.1316 -0.1426</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Female </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n     me      se     dif   lower   upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 52 10.626 0.07883  0.0000      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 26 10.176 0.18415 -0.4464 -0.7538 -0.1390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C  4  9.761 0.28635 -0.8617 -1.5245 -0.1989</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p trend: 0.36299 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Interaction SNP and SNP (codominant and codominant)</span></span></span>
<span class="r-in"><span><span class="fu">association</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span><span class="op">~</span><span class="va">snp10001</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">snp10002</span><span class="op">)</span><span class="op">+</span><span class="va">blood.pre</span>, </span></span>
<span class="r-in"><span>          data<span class="op">=</span><span class="va">datSNP</span>, model<span class="op">=</span><span class="st">"codominant"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 3 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       SNP: snp10001  adjusted by: blood.pre </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Interaction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           C/C             dif   lower   upper      A/C              dif   lower</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 38 10.591 0.1004  0.00000      NA      NA 49 10.66 0.07051  0.06403 -0.2133</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 24 10.502 0.1622 -0.08375 -0.4189  0.2514 29 10.23 0.13823 -0.36600 -0.6821</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 12  9.929 0.2439 -0.66157 -1.0860 -0.2371  0  0.00 0.00000       NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        upper n    me  A/A    dif   lower  upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T  0.34134 5 10.78 0.15 0.1799 -0.4327 0.7925</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T -0.04993 0  0.00 0.00     NA      NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C       NA 0  0.00 0.00     NA      NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p interaction: 0.13332 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  factor(snp10002) within snp10001 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n    me      se     dif   lower  upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 38 10.59 0.10044 0.00000      NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/C 49 10.66 0.07051 0.06403 -0.2133 0.3413</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/A  5 10.78 0.14998 0.17988 -0.4327 0.7925</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n    me     se     dif   lower   upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 24 10.50 0.1622  0.0000      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/C 29 10.23 0.1382 -0.2823 -0.6373 0.07276</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/A  0  0.00 0.0000      NA      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n    me     se dif lower upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 12 9.929 0.2439   0    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/C  0 0.000 0.0000  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/A  0 0.000 0.0000  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p trend: 0.41739 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  snp10001 within factor(snp10002) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n     me     se      dif   lower   upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 38 10.591 0.1004  0.00000      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 24 10.502 0.1622 -0.08375 -0.4189  0.2514</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 12  9.929 0.2439 -0.66157 -1.0860 -0.2371</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/C </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n    me      se   dif   lower   upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 49 10.66 0.07051  0.00      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 29 10.23 0.13823 -0.43 -0.7304 -0.1296</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C  0  0.00 0.00000    NA      NA      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A/A </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     n    me   se dif lower upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 5 10.78 0.15   0    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 0  0.00 0.00  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 0  0.00 0.00  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p trend: 0.097576 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Interaction SNP and SNP (dominant and recessive)</span></span></span>
<span class="r-in"><span><span class="fu">association</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span><span class="op">~</span><span class="va">snp10001</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="inheritance.html">recessive</a></span><span class="op">(</span><span class="va">snp100019</span><span class="op">)</span><span class="op">)</span><span class="op">+</span><span class="va">blood.pre</span>, </span></span>
<span class="r-in"><span>          data<span class="op">=</span><span class="va">datSNP</span>, model<span class="op">=</span><span class="st">"dominant"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 3 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=2) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'varlist' has changed (from nvar=3) to new 4 after EncodeVars() -- should no longer happen!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       SNP: snp10001  adjusted by: blood.pre </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Interaction </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        G/G-C/G             dif   lower    upper      C/C             dif</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 60  10.650 0.06314  0.0000      NA       NA 32 10.61 0.1117 -0.03823</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 53  10.351 0.10612 -0.2961 -0.5393 -0.05294  0  0.00 0.0000       NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 12   9.929 0.24389 -0.7191 -1.1257 -0.31256  0  0.00 0.0000       NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       lower  upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T -0.3196 0.2432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T      NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C      NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p interaction: NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  factor(recessive(snp100019)) within snp10001 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          n    me      se      dif   lower  upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G/G-C/G 60 10.65 0.06314  0.00000      NA     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C     32 10.61 0.11174 -0.03823 -0.3196 0.2432</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          n    me     se dif lower upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G/G-C/G 53 10.35 0.1061   0    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C      0  0.00 0.0000  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          n    me     se dif lower upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G/G-C/G 12 9.929 0.2439   0    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C      0 0.000 0.0000  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p trend: NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  snp10001 within factor(recessive(snp100019)) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G/G-C/G </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n     me      se     dif   lower    upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 60 10.650 0.06314  0.0000      NA       NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T 53 10.351 0.10612 -0.2961 -0.5393 -0.05294</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C 12  9.929 0.24389 -0.7191 -1.1257 -0.31256</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      n    me     se dif lower upper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T/T 32 10.61 0.1117   0    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/T  0  0.00 0.0000  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C/C  0  0.00 0.0000  NA    NA    NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> p trend: NA </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Victor Moreno, Juan R Gonzalez, Dolors Pelegri.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

