<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ SNPassoc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SNPassoc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0-17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/SNPassoc.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/brgelab" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/isglobal-brge/SNPassoc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="association.html">association()</a></code> 
        </dt>
        <dd>Association analysis between a single SNP and a given phenotype</dd>
      </dl><dl><dt>
          
          <code><a href="asthma.html">asthma</a></code> 
        </dt>
        <dd>SNP data on asthma case-control study</dd>
      </dl><dl><dt>
          
          <code><a href="BonferroniSig.html">Bonferroni.sig()</a></code> 
        </dt>
        <dd>Bonferroni correction of p values</dd>
      </dl><dl><dt>
          
          <code><a href="dscore.html">dscore()</a></code> <code><a href="dscore.html">pscore()</a></code> 
        </dt>
        <dd>Exact distribution of a genetic score</dd>
      </dl><dl><dt>
          
          <code><a href="GenomicControl.html">GenomicControl()</a></code> 
        </dt>
        <dd>Population substructure</dd>
      </dl><dl><dt>
          
          <code><a href="getGeneSymbol.html">getGeneSymbol()</a></code> 
        </dt>
        <dd>Get gene symbol from a list of SNPs</dd>
      </dl><dl><dt>
          
          <code><a href="getNiceTable.html">getNiceTable()</a></code> 
        </dt>
        <dd>Get Latex output</dd>
      </dl><dl><dt>
          
          <code><a href="getSignificantSNPs.html">getSignificantSNPs()</a></code> 
        </dt>
        <dd>Extract significant SNPs from an object of class 'WGassociation'</dd>
      </dl><dl><dt>
          
          <code><a href="haplointeraction.html">haplo.interaction()</a></code> 
        </dt>
        <dd>Haplotype interaction with a covariate</dd>
      </dl><dl><dt>
          
          <code><a href="HapMap.html">HapMap</a></code> 
        </dt>
        <dd>SNPs from HapMap project</dd>
      </dl><dl><dt>
          
          <code><a href="HapMap.SNPs.pos.html">HapMap.SNPs.pos</a></code> 
        </dt>
        <dd>SNPs from HapMap project</dd>
      </dl><dl><dt>
          
          <code><a href="inheritance.html">codominant()</a></code> <code><a href="inheritance.html">dominant()</a></code> <code><a href="inheritance.html">recessive()</a></code> <code><a href="inheritance.html">overdominant()</a></code> <code><a href="inheritance.html">additive()</a></code> 
        </dt>
        <dd>Collapsing (or recoding) genotypes into different categories (generally two) depending on a given genetic mode of inheritance</dd>
      </dl><dl><dt>
          
          <code><a href="int.html">int()</a></code> 
        </dt>
        <dd>Identify interaction term</dd>
      </dl><dl><dt>
          
          <code><a href="interactionPval.html">interactionPval()</a></code> 
        </dt>
        <dd>Two-dimensional SNP analysis for association studies</dd>
      </dl><dl><dt>
          
          <code><a href="intervals.html">intervals()</a></code> 
        </dt>
        <dd>Print ORs and 95% confidence intervals for an object of class 'haplo.glm'</dd>
      </dl><dl><dt>
          
          <code><a href="LD.html">LD()</a></code> <code><a href="LD.html">LDplot()</a></code> <code><a href="LD.html">LDtable()</a></code> 
        </dt>
        <dd>max-statistic for a 2x3 table</dd>
      </dl><dl><dt>
          
          <code><a href="makegeno.html">make.geno()</a></code> 
        </dt>
        <dd>Create a group of locus objects from some SNPs, assign to 'model.matrix' class.</dd>
      </dl><dl><dt>
          
          <code><a href="maxstat.html">maxstat()</a></code> 
        </dt>
        <dd>max-statistic for a 2x3 table</dd>
      </dl><dl><dt>
          
          <code><a href="odds.html">odds()</a></code> 
        </dt>
        <dd>Extract odds ratios, 95% CI and pvalues</dd>
      </dl><dl><dt>
          
          <code><a href="permTest.html">permTest()</a></code> 
        </dt>
        <dd>Permutation test analysis</dd>
      </dl><dl><dt>
          
          <code><a href="plotMissing.html">plotMissing()</a></code> 
        </dt>
        <dd>Plot of missing genotypes</dd>
      </dl><dl><dt>
          
          <code><a href="plotWGassociation.html">plot(<i>&lt;WGassociation&gt;</i>)</a></code> 
        </dt>
        <dd>Function to plot -log p values from an object of class 'WGassociation'</dd>
      </dl><dl><dt>
          
          <code><a href="qqpval.html">qqpval()</a></code> 
        </dt>
        <dd>Functions for inspecting population substructure</dd>
      </dl><dl><dt>
          
          <code><a href="related.html">related()</a></code> 
        </dt>
        <dd>Get related samples</dd>
      </dl><dl><dt>
          
          <code><a href="resHapMap.html">resHapMap</a></code> 
        </dt>
        <dd>SNPs from HapMap project</dd>
      </dl><dl><dt>
          
          <code><a href="scanWGassociation.html">scanWGassociation</a></code> 
        </dt>
        <dd>Whole genome association analysis</dd>
      </dl><dl><dt>
          
          <code><a href="setupSNP.html">setupSNP()</a></code> 
        </dt>
        <dd>Convert columns in a dataframe to class 'snp'</dd>
      </dl><dl><dt>
          
          <code><a href="snp.html">snp()</a></code> <code><a href="snp.html">is.snp()</a></code> <code><a href="snp.html">as.snp()</a></code> <code><a href="snp.html">additive(<i>&lt;snp&gt;</i>)</a></code> 
        </dt>
        <dd>SNP object</dd>
      </dl><dl><dt>
          
          <code><a href="SNPs.info.pos.html">SNPs.info.pos</a></code> 
        </dt>
        <dd>SNPs in a case-control study</dd>
      </dl><dl><dt>
          
          <code><a href="SNPs.html">SNPs</a></code> 
        </dt>
        <dd>SNPs in a case-control study</dd>
      </dl><dl><dt>
          
          <code><a href="sortSNPs.html">sortSNPs()</a></code> 
        </dt>
        <dd>Sort a vector of SNPs by genomic position</dd>
      </dl><dl><dt>
          
          <code><a href="tableHWE.html">tableHWE()</a></code> 
        </dt>
        <dd>Test for Hardy-Weinberg Equilibrium</dd>
      </dl><dl><dt>
          
          <code><a href="Tablemeanse.html">Table.mean.se()</a></code> 
        </dt>
        <dd>Descriptive sample size, mean, and standard error</dd>
      </dl><dl><dt>
          
          <code><a href="TableNPer.html">Table.N.Per()</a></code> 
        </dt>
        <dd>Descriptive sample size and percentage</dd>
      </dl><dl><dt>
          
          <code><a href="WGassociation.html">WGassociation()</a></code> 
        </dt>
        <dd>Whole genome association analysis</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Victor Moreno, Juan R Gonzalez, Dolors Pelegri.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

