<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="codominant function recodifies a variable having genotypes depending on the allelic frequency
             in descending order.              
dominant, recessive, and overdominant functions collapse the three categories of a given SNP 
            into two categories as follows: Let 'AA', 'Aa', and 'aa' be the three genotypes. After determining
            the most frequent allele (let's suppose that 'A' is the major allele) the functions return a vector
            with to categories as follows. dominant: 'AA' and 'Aa-aa'; recessive: 'AA-Aa' and 'aa'; 
            overdominant: 'AA-aa' vs 'Aa'.             
additive function creates a numerical variable, 1, 2, 3 corresponding to the three genotypes sorted
            out by descending allelic frequency (this model is referred as log-additive)."><title>Collapsing (or recoding) genotypes into different categories (generally two) depending on a given genetic mode of inheritance — inheritance • SNPassoc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Collapsing (or recoding) genotypes into different categories (generally two) depending on a given genetic mode of inheritance — inheritance"><meta property="og:description" content="codominant function recodifies a variable having genotypes depending on the allelic frequency
             in descending order.              
dominant, recessive, and overdominant functions collapse the three categories of a given SNP 
            into two categories as follows: Let 'AA', 'Aa', and 'aa' be the three genotypes. After determining
            the most frequent allele (let's suppose that 'A' is the major allele) the functions return a vector
            with to categories as follows. dominant: 'AA' and 'Aa-aa'; recessive: 'AA-Aa' and 'aa'; 
            overdominant: 'AA-aa' vs 'Aa'.             
additive function creates a numerical variable, 1, 2, 3 corresponding to the three genotypes sorted
            out by descending allelic frequency (this model is referred as log-additive)."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SNPassoc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0-17</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/SNPassoc.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/brgelab" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/isglobal-brge/SNPassoc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Collapsing (or recoding) genotypes into different categories (generally two) depending on a given genetic mode of inheritance</h1>
      
      <div class="d-none name"><code>inheritance.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>codominant function recodifies a variable having genotypes depending on the allelic frequency
             in descending order. <br></p>             
<p>dominant, recessive, and overdominant functions collapse the three categories of a given SNP 
            into two categories as follows: Let 'AA', 'Aa', and 'aa' be the three genotypes. After determining
            the most frequent allele (let's suppose that 'A' is the major allele) the functions return a vector
            with to categories as follows. dominant: 'AA' and 'Aa-aa'; recessive: 'AA-Aa' and 'aa'; 
            overdominant: 'AA-aa' vs 'Aa'. <br></p>            
<p>additive function creates a numerical variable, 1, 2, 3 corresponding to the three genotypes sorted
            out by descending allelic frequency (this model is referred as log-additive).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">codominant</span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span>
<span><span class="fu">dominant</span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span>
<span><span class="fu">recessive</span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span>
<span><span class="fu">overdominant</span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span>
<span><span class="fu">additive</span><span class="op">(</span><span class="va">o</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>o</dt>
<dd><p>categorical covariate having genotypes</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A snp object collapsing genotypes into different categories depending on a given genetic mode of inheritance</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SNPs</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">dominant</span><span class="op">(</span><span class="fu"><a href="snp.html">snp</a></span><span class="op">(</span><span class="va">SNPs</span><span class="op">$</span><span class="va">snp10001</span>,sep<span class="op">=</span><span class="st">""</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1] T/T     T/T     T/T     C/T-C/C T/T     T/T     T/T     T/T     C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [10] T/T     C/T-C/C C/T-C/C C/T-C/C T/T     T/T     T/T     C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [19] T/T     T/T     T/T     C/T-C/C C/T-C/C T/T     T/T     C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [28] T/T     T/T     C/T-C/C T/T     T/T     C/T-C/C T/T     C/T-C/C C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [37] C/T-C/C C/T-C/C C/T-C/C T/T     T/T     T/T     T/T     C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [46] C/T-C/C C/T-C/C C/T-C/C C/T-C/C T/T     T/T     C/T-C/C C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [55] T/T     T/T     T/T     T/T     C/T-C/C T/T     T/T     T/T     T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [64] T/T     T/T     T/T     C/T-C/C T/T     T/T     C/T-C/C C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [73] T/T     C/T-C/C T/T     T/T     T/T     T/T     T/T     C/T-C/C C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [82] T/T     T/T     T/T     C/T-C/C T/T     T/T     C/T-C/C C/T-C/C C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [91] T/T     T/T     C/T-C/C T/T     T/T     T/T     T/T     T/T     C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [100] T/T     C/T-C/C C/T-C/C C/T-C/C T/T     T/T     C/T-C/C C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [109] T/T     C/T-C/C T/T     C/T-C/C C/T-C/C T/T     C/T-C/C C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [118] T/T     C/T-C/C C/T-C/C C/T-C/C T/T     C/T-C/C T/T     T/T     T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [127] T/T     T/T     C/T-C/C T/T     C/T-C/C C/T-C/C T/T     C/T-C/C C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [136] T/T     C/T-C/C T/T     T/T     C/T-C/C C/T-C/C C/T-C/C T/T     T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [145] T/T     T/T     C/T-C/C T/T     T/T     T/T     T/T     C/T-C/C T/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [154] C/T-C/C C/T-C/C C/T-C/C C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genotypes: T/T C/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Alleles:  </span>
<span class="r-in"><span><span class="fu">overdominant</span><span class="op">(</span><span class="fu"><a href="snp.html">snp</a></span><span class="op">(</span><span class="va">SNPs</span><span class="op">$</span><span class="va">snp10001</span>,sep<span class="op">=</span><span class="st">""</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1] T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [10] T/T-C/C C/T     T/T-C/C C/T     T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [19] T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C T/T-C/C C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [28] T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C C/T     C/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [37] T/T-C/C C/T     C/T     T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [46] T/T-C/C C/T     C/T     C/T     T/T-C/C T/T-C/C C/T     C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [55] T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C T/T-C/C T/T-C/C T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [64] T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T     C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [73] T/T-C/C C/T     T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T     C/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [82] T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [91] T/T-C/C T/T-C/C C/T     T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [100] T/T-C/C C/T     C/T     C/T     T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [109] T/T-C/C C/T     T/T-C/C C/T     C/T     T/T-C/C C/T     C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [118] T/T-C/C C/T     C/T     C/T     T/T-C/C T/T-C/C T/T-C/C T/T-C/C T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [127] T/T-C/C T/T-C/C C/T     T/T-C/C C/T     T/T-C/C T/T-C/C C/T     C/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [136] T/T-C/C C/T     T/T-C/C T/T-C/C C/T     C/T     C/T     T/T-C/C T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [145] T/T-C/C T/T-C/C C/T     T/T-C/C T/T-C/C T/T-C/C T/T-C/C C/T     T/T-C/C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [154] C/T     C/T     C/T     C/T    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genotypes: T/T-C/C C/T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Alleles:  </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Victor Moreno, Juan R Gonzalez, Dolors Pelegri.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

